Firestore/Source/API/FIRVectorValue.mm

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#include <vector>
+
+#include "Firestore/Source/Public/FirebaseFirestore/FIRVectorValue.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface FIRVectorValue () {
+  /** Internal vector representation */
+  std::vector<double> _internalValue;
```

Comment:
Since this is a private instance variable, I'm wondering if it makes sense to make use an NSArray as the backing type instead of dropping down here into C++ at this level? 

I think doing so would consolidate the similar logic in `-[FIRVectorValue array]` and `- [FIRVectorValue initWithArray:]` so that the input array is copied and traversed, and each value goes into a new array meant to back this class.

Reply:
nice cleanup. fixed.